addons/stock/models/stock_picking.py

```diff
@@ -1026,7 +1026,7 @@ def _put_in_pack(self, move_line_ids):
             package = self.env['stock.quant.package'].create({})
             for ml in move_line_ids:
                 if float_compare(ml.qty_done, ml.product_uom_qty,
-                                 precision_rounding=ml.product_uom_id.rounding) >= 0:
+                                 precision_rounding=ml.product_uom_id.rounding) >= 0 or ml.qty_done == 0.0:
```

Comment:
this is ugly as hell + you don't even use float_compare

Reply:
in put_in_pack, either:
- you have some quantities done anywhere and you keep the old behaviour
- you don't have any quantities done at all BUT you some moves are reserved, then set the qty done as the qty reserved on the move line and call the old code

Reply:
Done